rlog/2024-04-20-rln-light-verifiers.mdx

```diff
@@ -0,0 +1,163 @@
+---
+title: 'Verifying RLN Proofs in Light Clients with Subtrees'
+date: 2024-04-20 12:00:00
+authors: p1ge0nh8er
+published: true
+slug: rln-light-verifiers
+categories: research
+
+toc_min_heading_level: 2
+toc_max_heading_level: 4
+---
+
+How resource-restricted devices can verify RLN proofs fast and efficiently.
+
+<!--truncate-->
+
+## Introduction
+
+Recommended previous reading: [Strengthening Anonymous DoS Prevention with Rate Limiting Nullifiers in Waku](https://vac.dev/rlog/rln-anonymous-dos-prevention).
+
+This post expands upon ideas described in the previous post,
+focusing on how resource-restricted devices can verify RLN proofs fast and efficiently.
+
+Previously, it was required to fetch all the memberships from the smart contract,
+construct the merkle tree locally,
+and derive the merkle root,
+which is subsequently used to verify RLN proofs.
+
+This process is not feasible for resource-restricted devices,
```

Comment:
Drop comma between 'devices' and 'since'

Reply:
addressed in 74ab90b